tensorflow-examples/src/main/java/org/tensorflow/model/examples/datasets/mnist/MnistDataset.java

```diff
@@ -14,29 +14,33 @@
  *  limitations under the License.
  *  =======================================================================
  */
-package org.tensorflow.model.examples.mnist.data;
+package org.tensorflow.model.examples.datasets.mnist;
 
-import static org.tensorflow.tools.ndarray.index.Indices.from;
-import static org.tensorflow.tools.ndarray.index.Indices.to;
-
-import java.io.DataInputStream;
-import java.io.IOException;
-import java.util.zip.GZIPInputStream;
+import org.tensorflow.model.examples.datasets.ImageBatch;
+import org.tensorflow.model.examples.datasets.ImageBatchIterator;
 import org.tensorflow.tools.Shape;
 import org.tensorflow.tools.buffer.DataBuffers;
 import org.tensorflow.tools.ndarray.ByteNdArray;
 import org.tensorflow.tools.ndarray.NdArrays;
 
-public class MnistDataset {
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.util.zip.GZIPInputStream;
 
+import static org.tensorflow.tools.ndarray.index.Indices.from;
+import static org.tensorflow.tools.ndarray.index.Indices.to;
+
+/** Common loader and data preprocessor for MNIST and FashionMNIST datasets. */
+public class MnistDataset {
   public static final int NUM_CLASSES = 10;
 
-  public static MnistDataset create(int validationSize) {
+  public static MnistDataset create(int validationSize, String trainingImagesArchive, String trainingLabelsArchive,
+                                    String testImagesArchive, String testLabelsArchive) {
     try {
-      ByteNdArray trainingImages = readArchive(TRAINING_IMAGES_ARCHIVE);
-      ByteNdArray trainingLabels = readArchive(TRAINING_LABELS_ARCHIVE);
-      ByteNdArray testImages = readArchive(TEST_IMAGES_ARCHIVE);
-      ByteNdArray testLabels = readArchive(TEST_LABELS_ARCHIVE);
+      ByteNdArray trainingImages = readArchive(trainingImagesArchive);
```

Comment:
Maybe there should be two methods here, one for MNIST and one for FashionMNIST? Then we don't have to carry around the string constants in all the classes that use them. This method can remain if people want to override it to load in the extra MNIST test data from last years NeurIPS.

Reply:
Maybe, but I've copied an existing approach, from my perspective for examples the best way to have them as a resources.